manifests/packages.pp

```diff
@@ -1,12 +1,20 @@
-class wireguard::packages {
-  if($::osfamily == 'debian') {
+# Install wireguard packages for kernel module and tools
+class wireguard::packages (
+  # Wireguard tools only work with the kernel module with the same version; take
+  # care if using `latest` as it may break the tools until the module has been
+  # reloaded
+  Variant[Enum['installed', 'latest', 'present'], String] $version = 'installed',
```

Comment:
I'm a bit unhappy with calling this `$version`. Maybe `$ensure` makes more sense, as it's already pretty common in some puppet resources.

Reply:
I'm fine with either. I'm not that fond of overloading `ensure` to set not just define whether a package should be present or not but also to control which version, but since it's common to use `ensure` -- let's keep it that way.